vpr/src/device/rr_graph_builder.h

```diff
@@ -0,0 +1,63 @@
+#ifndef RR_GRAPH_BUILDER_H
+#define RR_GRAPH_BUILDER_H
+
+#include "rr_graph_storage.h"
+#include "rr_spatial_lookup.h"
+
+/* A data structure allows data modification on a routing resource graph 
+ *
+ * Note that the builder does not own the storage
+ * It serves a virtual protocol for
+ * - node_storage: store the node list 
+ * - node_lookup: store a fast look-up for the nodes
+ *
+ * Note:
+ * - This is the only data structre allowed to modify a routing resource graph
+ *
+ */
+class RRGraphBuilder {
+    /****************
+     * Constructors
+     ****************/
+  public:
+    RRGraphBuilder(t_rr_graph_storage* node_storage,
+                   RRSpatialLookup* node_lookup);
+
+    /* Disable copy constructors
```

Comment:
This looks like a lot of comment for something fairly common.
I think the disabled copy constructor and assignment operator can be there without comment as they speak for themselve.

Reply:
I left the comment about the copy constructor is because it is the first time I used these C++ syntax. 
I did so because I assume that there could be many guys do not know the syntax as well. 
On the other sides, we cannot expect every developer is an expert of a C++ when reading the source codes (considering VPR is a research project, it is perhaps the first C++ project that students touched).
Some comments may be appreciated for these non-popular syntax.
Would you mind share your experience? 
My idea is to find a good trade-off in the comments between entry-level engineers and experts.

Reply:
No worries, leave it as you feel comfortable.

As for the additional explanation
```
 This is to avoid any duplication of the object
 as it is only interface allowed to modify routing resource graph
```

I'd probably emphasize that we want to avoid an accidental copy not because of the interface situation, but because (once the builder actually owns the data), it will be a very expensive operation, copying gigabytes of data. If we prevent copying, then accidental 'pass-by-value' are immediately caught by the compiler for instance.

Reply:
No problem. I think it is more accurate. Changed.